src/ux/table.ts

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { inspect } from 'node:util';
+import ansis from 'ansis';
+import { orderBy } from 'natural-orderby';
+import sliceAnsi from 'slice-ansi';
+import sw from 'string-width';
+import { ux } from '@oclif/core';
+
+function sumBy<T>(arr: T[], fn: (i: T) => number): number {
+  return arr.reduce((sum, i) => sum + fn(i), 0);
+}
+
+function capitalize(s: string): string {
+  return s ? s.charAt(0).toUpperCase() + s.slice(1).toLowerCase() : '';
+}
+
+function termwidth(stream: NodeJS.WriteStream): number {
+  if (!stream.isTTY) {
+    return 80;
+  }
+
+  const width = stream.getWindowSize()[0];
+  if (width < 1) {
+    return 80;
+  }
+
+  if (width < 40) {
+    return 40;
+  }
+
+  return width;
+}
+
+const stdtermwidth = Number.parseInt(process.env.OCLIF_COLUMNS!, 10) || termwidth(process.stdout);
+
```

Comment:
checking the type first instead of relying on `!` and NaN's behavior is clearer
```suggestion
const stdtermwidth =
  typeof process.env.OCLIF_COLUMNS === 'string' ? Number.parseInt(process.env.OCLIF_COLUMNS, 10) : termwidth(process.stdout);
```